pkg/provider/pulumi/pulumi.go

```diff
@@ -0,0 +1,90 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pulumi
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	esc "github.com/pulumi/esc/cmd/esc/cli/client"
+	corev1 "k8s.io/api/core/v1"
+
+	esv1beta1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
+)
+
+type client struct {
+	escClient    esc.Client
+	environment  string
+	organization string
+	duration     time.Duration
+}
+
+var _ esv1beta1.SecretsClient = &client{}
+
+func (c *client) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	x, _, err := c.escClient.OpenEnvironment(ctx, c.organization, c.environment, c.duration)
+	if err != nil {
+		return nil, err
+	}
+	value, err := c.escClient.GetOpenProperty(ctx, c.organization, c.environment, x, ref.Key)
+	if err != nil {
+		return nil, err
+	}
+	return []byte(value.ToString(false)), nil
```

Comment:
The `value.ToString(true)` has a quirk where it renders a map as `%q=%q` formatted string which is not very useful.

The utils package `github.com/external-secrets/external-secrets/pkg/utils` provides a function to return a `[]byte` representation of a `interface{}` type which would be helpful in this case:

```suggestion
	return utils.GetByteValue(value.ToJSON(false))
```

Given this environment json:
```json
{
  "environmentVariables": {
    "EXAMPLE_SETTING": "changeme"
  },
  "example": {
    "setting": "changeme"
  }
}
```

I can define the following keys and receive these values:
| key | value (value.ToString()) | value (GetByteValue) | 
|-----|-------|-----|
|   `""`  |  `""`     | `""` |
|   `example`  |   `"setting"="changeme"`    | `{"setting":"changeme"}` |
|   `example.setting`  |  `changeme`     | `changeme` |